### ISSUE TITLE
Documentation is generated wrongly for PhysicBody AxisLock methods 

### ISSUE THREAD

**Issue by kubecz3k** [godotengine/godot#16082]
**Godot version:**
Godot 3.0 RC3 

**OS/device including version:**
Ubuntu 17.04

**Issue description:**
Overview: 
~*1. BodyAxis is not exposed*~
*2. Buildin help has wrong definitions of `set_axis_lock` and `get_axis_lock`*

Full description:
It seems we have missing/wrong bindings around `body axis lock` api. I was trying to use `RigidBody.set_axis_lock` but I'm unable to figure out how to do this. According to buildin help there is one parameter of `get/set_axis_lock`: 
![axis_lock_screen](https://user-images.githubusercontent.com/6129594/35437818-5aa55e9e-0294-11e8-9f4c-ea50c773664b.png)
But when looking at `physic_body.cpp` at the binded method there are in fact 2 parameters:
`void RigidBody::set_axis_lock(PhysicsServer::BodyAxis p_axis, bool p_lock) ` 

Similar api is also visible in `PhysicServer` build in help:
 `void body_set_axis_lock( RID body, BodyAxis axis, bool lock )`
~Also it can be seen that `BodyAxis` enum is a second parameter, but this enum is not exposed anywhere, so it can't be used from GdScript.~
Update in comment: https://github.com/godotengine/godot/issues/16082#issuecomment-362195944

**Steps to reproduce:**
1. Open `RigidBody` buildin help.
2. Check `set_axis_lock` methods, they should have 2 parameters: (BodyAxis p_axis, bool p_lock)
3. Try to find `BodyAxis` enum from Gdscript code or find it's definition in Godot help

**Comment by kubecz3k** [comment]
@AndreaCatania you made the new api, could you read the issue to check if I haven't miss anything? 
Thank you!

**Comment by AndreaCatania** [comment]
I've the function `set_axis_lock` and `get_axis_lock`, The declaration is the following:

	ClassDB::bind_method(D_METHOD("set_axis_lock", "axis", "lock"), &RigidBody::set_axis_lock);
	ClassDB::bind_method(D_METHOD("get_axis_lock", "axis"), &RigidBody::get_axis_lock);

Then I've 6 parameters that are exposed to the editor and the declration is this: 

```
ADD_GROUP("Axis Lock", "axis_lock_");
ADD_PROPERTYI(PropertyInfo(Variant::BOOL, "axis_lock_linear_x"), "set_axis_lock", "get_axis_lock", PhysicsServer::BODY_AXIS_LINEAR_X);
ADD_PROPERTYI(PropertyInfo(Variant::BOOL, "axis_lock_linear_y"), "set_axis_lock", "get_axis_lock", PhysicsServer::BODY_AXIS_LINEAR_Y);
ADD_PROPERTYI(PropertyInfo(Variant::BOOL, "axis_lock_linear_z"), "set_axis_lock", "get_axis_lock", PhysicsServer::BODY_AXIS_LINEAR_Z);
ADD_PROPERTYI(PropertyInfo(Variant::BOOL, "axis_lock_angular_x"), "set_axis_lock", "get_axis_lock", PhysicsServer::BODY_AXIS_ANGULAR_X);
ADD_PROPERTYI(PropertyInfo(Variant::BOOL, "axis_lock_angular_y"), "set_axis_lock", "get_axis_lock", PhysicsServer::BODY_AXIS_ANGULAR_Y);
ADD_PROPERTYI(PropertyInfo(Variant::BOOL, "axis_lock_angular_z"), "set_axis_lock", "get_axis_lock", PhysicsServer::BODY_AXIS_ANGULAR_Z);
```

As you can see I'm using the function set_axis_lock and get_axis_lock to handle the parameters.

---

> ![](https://user-images.githubusercontent.com/6129594/35437818-5aa55e9e-0294-11e8-9f4c-ea50c773664b.png)
> But when looking at physic_body.cpp at the binded method there are in fact 2 parameters:
> void RigidBody::set_axis_lock(PhysicsServer::BodyAxis p_axis, bool p_lock)


This doc is correct since you need only one parameter to use set_lock_linear_y

The real issue here is that for some reason the functions set_axis_lock and get_axis_lock are not correctly exposed in Public Methods (but are threated as hidden methods similarly when I use "_" "_method_name")

Probably there's something more to set in order to expose these, I'm serching it

**Comment by AndreaCatania** [comment]
The same thing happen in camera2D, try to search for `set_limit` in script class. Is it the wanted behavior?

**Comment by kubecz3k** [comment]
We were discussing with @AndreaCatania on IRC and our consensus is that this issue is a problem with DOC generator. The right DOC should be something similar to this: `set_axis_lock(PhysicsServer::BODY_AXIS_LINEAR_Y, value);`

**Comment by AndreaCatania** [comment]
Is this fixed?

**Comment by reduz** [comment]
I may have fixed this, can you check?

**Comment by akien-mga** [comment]
Doesn't seem fixed to me in d9a96878. `PhysicsServer::body_set_axis_lock` is properly shown in the documentation, but not `RigidBody::set_axis_lock`.

**Comment by KoBeWi** [comment]
As of 3.2, the setter methods with additional parameters are exposed and documented, so you can find how to use them. Although we could make it more convenient if these setters were clickable from the property description.
